source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -37,9 +44,9 @@ public MessageExchangeStream(Stream stream, ILog log)
 
         public void IdentifyAsClient()
         {
+            log.Write(EventType.Diagnostic, $"Sent: {MxClient} {currentVersion}");
```

Comment:
This logging should be extracted to a method (that logs (if enabled) and writes to the stream)